lib/qesapdeployment.pm

```diff
@@ -87,6 +91,56 @@ sub qesap_create_folder_tree {
     assert_script_run("mkdir -p $paths{deployment_dir}", quiet => 1);
 }
 
+=head3 qesap_get_variables
+
+    Scans yaml config for '%OPENQA_VARIABLE%' placeholders and searches for values in OpenQA defined variables.
+    Returns hash with openqa variable key/value pairs.
+=cut
+
+sub qesap_get_variables {
+    my %paths = qesap_get_file_paths();
+    my $yaml_file = $paths{'qesap_conf_src'};
+    my %variables;
+    my $grep_cmd = "grep -v '#' | grep -oE %[A-Z0-9_]*% | sed s/%//g";
+    my $cmd = join(" ", "curl -s -L", $yaml_file, "|", $grep_cmd);
+
+    for my $variable (split(" ", script_output($cmd))) {
+        $variables{$variable} = get_required_var($variable);
+    }
+    return \%variables;
+}
+
+=head3 qesap_create_ansible_section
+
+    Writes "ansible" section into yaml config file.
+    $args{ansible_section} defines section(key) name.
+    $args{section_content} defines content of names section.
+        Example:
+            @playbook_list = ("pre-cluster.yaml", "cluster_sbd_prep.yaml");
+            qesap_create_ansible_section(ansible_section=>'create', section_content=>\@playbook_list);
+
+=cut
+
+sub qesap_create_ansible_section {
+    my (%args) = @_;
+    my $ypp = YAML::PP->new;
+    my $section = $args{ansible_section};
+    my $content = $args{section_content};
+    my %paths = qesap_get_file_paths();
+    my $yaml_config_path = $paths{qesap_conf_trgt};
+    die("Yaml config file '$yaml_config_path' does not exist.") if assert_script_run("test $yaml_config_path") > 0;
+    my $raw_file = script_output("cat $yaml_config_path");
+    my $yaml_data = $ypp->load_string($raw_file);
+
+    $yaml_data->{ansible}{$section} = $content;
```

Comment:
Generic consideration (no need any code change in this PR): this function introduces a openQA test code dependency with the internal conf.yaml of qe-sap-deployment structure. Now this function and the code of the caller 

```
my $section = $args{ansible_section};
```

guess on the fact that the YAML has an **ansible:** section and the caller guess that the **ansible:** has a **create:** (or other) section in it intended to be a sequence of playbooks. Before this PR the dependency was only in the files in  data/sles4sap/qe_sap_deployment and the gluescript uses **apiver:**
 in them to try to keep any dependency as transparent as possible.

Reply:
You cannot avoid this dependency. Yaml files are very static and I am not sure how would you like to have granular control over qesap settings from openQA side. Remeber there will be a lot of scenarios and having multiple almost identical yaml files is not ideal either. %VAR% method is really limited. There is a problem if you introduce a simple list like adding list of hana media or list of playbooks.
Maybe we should stop changing yaml format constantly? I see this as much bigger problem already.